lib/internal/test_runner/runner.js

```diff
@@ -248,14 +220,79 @@ class FileTest extends Test {
       this.#buffer = [];
     }
   }
-  addToReport(ast) {
-    this.#accumulateReportItem(ast);
+  addToReport(item) {
+    this.#accumulateReportItem(item);
     if (!this.isClearToSend()) {
-      ArrayPrototypePush(this.#buffer, ast);
+      ArrayPrototypePush(this.#buffer, item);
       return;
     }
     this.#drainBuffer();
-    this.#handleReportItem(ast);
+    this.#handleReportItem(item);
+  }
+  parseMessage(readData) {
+    if (readData.length === 0) return;
+
+    ArrayPrototypePush(this.#messageBuffer, readData);
+    this.#messageBufferSize += readData.length;
+
+    // Index 0 should always be present because we just pushed data into it.
+    let messageBufferHead = this.#messageBuffer[0];
+    let headerIndex = messageBufferHead.indexOf(v8Header);
+    let nonSerialized = Buffer.alloc(0);
+
+    while (messageBufferHead && headerIndex !== kSerializedSizeHeader) {
+      const nonSerializedData = headerIndex === -1 ?
+        messageBufferHead :
+        messageBufferHead.slice(0, headerIndex - kSerializedSizeHeader);
```

Comment:
if this is a typed array, this should have a primordial also

Reply:
same - this is a Buffer